Unable to run yarn install

Unable to run yarn install from develop branch.

## Expected Behavior

The project installation should work.

## Current Behavior
```bash
yarn install v1.22.5
[1/5] Validating package.json...
[2/5] Resolving packages...
success Already up-to-date.
$ yarn clean && yarn build
yarn run v1.22.5
$ rm -rf dist/npm
Done in 0.14s.
yarn run v1.22.5
$ yarn build:schemas && yarn build:lib && yarn build:web
$ mkdir -p dist/npm/common && cp -r src/common/schemas dist/npm/common/
$ tsc --build
snippets/src/reliableTransactionSubmission.ts:144:20 - error TS2794: Expected 1 arguments, but got 0. Did you forget to include 'void' in your type argument to 'Promise'?

144             return resolve()
                       ~~~~~~~~~

  node_modules/typescript/lib/lib.es2015.promise.d.ts:33:34
    33     new <T>(executor: (resolve: (value: T | PromiseLike<T>) => void, reject: (reason?: any) => void) => void): Promise<T>;
                                        ~~~~~~~~~~~~~~~~~~~~~~~~~
    An argument for 'value' was not provided.


Found 1 error.

error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

## Steps to Reproduce

1. git clone git@github.com:ripple/ripple-lib.git
2. yarn install

## Context

It seems the issue was introduced by the update of TypeScipt https://github.com/ripple/ripple-lib/blob/6938773a2222979855ef71530284465575127a5a/package.json#L59

--- a/snippets/src/reliableTransactionSubmission.ts
+++ b/snippets/src/reliableTransactionSubmission.ts
@@ -104,7 +104,7 @@ async function performPayments(payments) {
     // Most of the time we'll get 'tesSUCCESS' or (after many submissions) 'terQUEUED'
     console.log(`tx ${i} - tentative: ${result.resultCode}`)
 
-    const txFinalizedPromise = new Promise((resolve) => {
+    const txFinalizedPromise = new Promise<void>((resolve) => {
       const ledgerClosedCallback = async (event: LedgerClosedEvent) => {
         let status
         try {

